packages/editor-ui/src/App.vue

```diff
@@ -1,3 +1,65 @@
+<script setup lang="ts">
+import { ref, computed, watch, onMounted } from 'vue';
+import { useRoute } from 'vue-router';
+import LoadingView from '@/views/LoadingView.vue';
+import BannerStack from '@/components/banners/BannerStack.vue';
+import AskAssistantChat from '@/components/AskAssistant/AskAssistantChat.vue';
+import Modals from '@/components/Modals.vue';
+import Telemetry from '@/components/Telemetry.vue';
+import AskAssistantFloatingButton from '@/components/AskAssistant/AskAssistantFloatingButton.vue';
+import { loadLanguage } from '@/plugins/i18n';
+import { useExternalHooks } from '@/composables/useExternalHooks';
+import { HIRING_BANNER, VIEWS } from '@/constants';
+import { useRootStore } from '@/stores/root.store';
+import { useAssistantStore } from './stores/assistant.store';
+import { useUIStore } from './stores/ui.store';
+import { useUsersStore } from './stores/users.store';
+import { useSettingsStore } from './stores/settings.store';
+import { useGlobalLinkActions } from '@/composables/useGlobalLinkActions';
+import { useHistoryHelper } from '@/composables/useHistoryHelper';
+import { useToast } from '@/composables/useToast';
+
+const route = useRoute();
+const rootStore = useRootStore();
+const assistantStore = useAssistantStore();
+const uiStore = useUIStore();
+const usersStore = useUsersStore();
+const settingsStore = useSettingsStore();
+
+const globalLinkActions = useGlobalLinkActions();
+const historyHelper = useHistoryHelper(route);
+const toast = useToast();
+const externalHooks = useExternalHooks();
+
+const loading = ref(true);
+const defaultLocale = computed(() => rootStore.defaultLocale);
+const isDemoMode = computed(() => route.name === VIEWS.DEMO);
+const showAssistantButton = computed(() => assistantStore.canShowAssistantButtons);
+
+watch(defaultLocale, (newLocale) => {
+	void loadLanguage(newLocale);
+});
+
+const logHiringBanner = () => {
+	if (settingsStore.isHiringBannerEnabled && !isDemoMode.value) {
+		console.log(HIRING_BANNER);
+	}
+};
+
+onMounted(async () => {
+	logHiringBanner();
+	void useExternalHooks().run('app.mount');
+	loading.value = false;
+});
+
+defineExpose({
+	...globalLinkActions,
+	...historyHelper,
+	...toast,
+	...externalHooks,
```

Comment:
Are these used somewhere via a reference to the app component?

Reply:
No, not by reference but we need to initialize them. Will remove the `defineExpose`